zingo-status/src/confirmation_status.rs

```diff
@@ -85,17 +89,19 @@ impl ConfirmationStatus {
     /// use zcash_primitives::consensus::BlockHeight;
     ///
     /// let status = ConfirmationStatus::Confirmed(10.into());
-    /// assert_eq!(status.is_confirmed_before_or_at(&8.into()), false);
+    /// assert_eq!(status.is_confirmed_before_or_at(&9.into()), false);
     ///
     /// let status = ConfirmationStatus::Broadcast(10.into());
     /// assert_eq!(status.is_confirmed_before_or_at(&10.into()), false);
     ///
     /// let status = ConfirmationStatus::Confirmed(10.into());
-    /// assert_eq!(status.is_confirmed_before_or_at(&12.into()), true);
+    /// assert_eq!(status.is_confirmed_before_or_at(&11.into()), true);
     /// ```
     pub fn is_confirmed_before_or_at(&self, comparison_height: &BlockHeight) -> bool {
         match self {
-            Self::Confirmed(self_height) => self_height <= comparison_height,
+            Self::Confirmed(self_height) => {
+                self.is_confirmed_before(comparison_height) || self_height == comparison_height
+            }
```

Comment:
this is a legibility improvement! i love it!

Reply:
It's a also a unification of sources of truth.